drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,599 @@
+{
+    dev->settings.state = state;
+}
+
+void sx127x_set_modem(sx127x_t *dev, sx127x_radio_modems_t modem)
+{
+    dev->settings.modem = modem;
+
+    switch (dev->settings.modem) {
+        case SX127X_MODEM_LORA:
+            sx127x_set_op_mode(dev, SX127X_RF_OPMODE_SLEEP);
+            sx127x_reg_write(dev, SX127X_REG_OPMODE,
+                             (sx127x_reg_read(dev, SX127X_REG_OPMODE) &
+                              SX127X_RF_LORA_OPMODE_LONGRANGEMODE_MASK) |
+                             SX127X_RF_LORA_OPMODE_LONGRANGEMODE_ON);
+
+            sx127x_reg_write(dev, SX127X_REG_DIOMAPPING1, 0x00);
+            sx127x_reg_write(dev, SX127X_REG_DIOMAPPING2, 0x10); /* DIO5=ClkOut */
+            break;
+
+        case SX127X_MODEM_FSK:
+            sx127x_set_op_mode(dev, SX127X_RF_OPMODE_SLEEP);
+            sx127x_reg_write(dev, SX127X_REG_OPMODE,
+                             (sx127x_reg_read(dev, SX127X_REG_OPMODE) &
+                              SX127X_RF_LORA_OPMODE_LONGRANGEMODE_MASK) |
+                             SX127X_RF_LORA_OPMODE_LONGRANGEMODE_OFF);
+
+            sx127x_reg_write(dev, SX127X_REG_DIOMAPPING1, 0x00);
+            break;
+        default:
+            break;
+    }
+}
+
+uint8_t sx127x_get_syncword(sx127x_t *dev)
+{
+    return sx127x_reg_read(dev, SX127X_REG_LR_SYNCWORD);
+}
+
+void sx127x_set_syncword(sx127x_t *dev, uint8_t syncword)
+{
+    sx127x_reg_write(dev, SX127X_REG_LR_SYNCWORD, syncword);
+}
+
+uint32_t sx127x_get_channel(sx127x_t *dev)
+{
+    return ((sx127x_reg_read(dev, SX127X_REG_FRFMSB) << 16) |
+            (sx127x_reg_read(dev, SX127X_REG_FRFMID) << 8) |
+            (sx127x_reg_read(dev, SX127X_REG_FRFLSB))) * SX127X_FREQ_STEP;
+
+}
+
+void sx127x_set_channel(sx127x_t *dev, uint32_t freq)
+{
+    /* Save current operating mode */
+    dev->settings.channel = freq;
+    uint8_t prev_mode = sx127x_reg_read(dev, SX127X_REG_OPMODE);
+
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_STANDBY);
+
+    freq = (uint32_t)((double) freq / (double) SX127X_FREQ_STEP);
+
+    /* Write frequency settings into chip */
+    sx127x_reg_write(dev, SX127X_REG_FRFMSB, (uint8_t)((freq >> 16) & 0xFF));
+    sx127x_reg_write(dev, SX127X_REG_FRFMID, (uint8_t)((freq >> 8) & 0xFF));
+    sx127x_reg_write(dev, SX127X_REG_FRFLSB, (uint8_t)(freq & 0xFF));
+
+    /* Restore previous operating mode */
+    sx127x_reg_write(dev, SX127X_REG_OPMODE, prev_mode);
+}
+
+uint32_t sx127x_get_time_on_air(sx127x_t *dev)
+{
+    uint32_t air_time = 0;
+
+    uint8_t pkt_len = dev->settings.time_on_air_pkt_len;
+    sx127x_radio_modems_t modem = dev->settings.modem;
+    switch (modem) {
+        case SX127X_MODEM_FSK:
+            break;
+
+        case SX127X_MODEM_LORA:
+        {
+            double bw = 0.0;
+
+            /* Note: When using LoRa modem only bandwidths 125, 250 and 500 kHz are supported. */
+            switch (dev->settings.lora.bandwidth) {
+                case 7: /* 125 kHz */
```

Comment:
Good catch. There's a `enum` type `sx127x_lora_bandwidth_t` defined in `sx127x.h` and it has to be a `SX127X_BW_125_KHZ` member from it and for following cases as well.